src/db/mod.rs

```diff
@@ -32,15 +32,19 @@ impl<'file> Database<'file> {
 
     /// Adds a new directory or increments its rank. Also updates its last accessed time.
     pub fn add<S: AsRef<str>>(&mut self, path: S, now: Epoch) {
+        self.add_raw(path, now, 1.0)
+    }
+
+    pub fn add_raw<S: AsRef<str>>(&mut self, path: S, last_accessed: Epoch, rank: Rank) {
```

Comment:
No need of the original function, just add this parameter to this original `add` function.